src/detect-engine-alert.c

```diff
@@ -276,6 +276,44 @@ static uint16_t AlertQueueExpand(DetectEngineThreadCtx *det_ctx)
     return new_cap;
 }
 
+static inline int PacketAlertSetContext(
+        DetectEngineThreadCtx *det_ctx, PacketAlert *pa, const Signature *s)
+{
+    pa->json_info = NULL;
+    if (det_ctx->json_content_len) {
+        /* We have some JSON attached in the current detection so let's try
+           to see if some need to be used for current signature. */
+        struct PacketContextData *current_json = pa->json_info;
+        if (current_json == NULL) {
+            current_json = SCCalloc(1, sizeof(struct PacketContextData));
+            if (current_json == NULL) {
+                /* Allocation error, let's return now */
+                return -1;
+            }
+            pa->json_info = current_json;
+        }
+        for (size_t i = 0; i < det_ctx->json_content_len; i++) {
```

Comment:
this is a bit weird. We call `PacketAlertSetContext` when we add a specific alert for a specific sid, but we do a linear search for the data belonging to the sig here. This appears to be needed because we only reset `json_content_len` at the start of a detection run, instead of between individual sigs inspected. Why is that?

Reply:
No need to cross signature here so I can definitely look at making it in smaller part.